common/build/eslint-plugin-fluid/src/rules/no-unchecked-record-access.js

```diff
@@ -515,3 +578,111 @@ function isStrictlyTypedVariable(typeAnnotation) {
 	// Consider any non-union type as strictly typed, except for 'any' and 'unknown'
 	return typeAnnotation.type !== "TSAnyKeyword" && typeAnnotation.type !== "TSUnknownKeyword";
 }
+
+// Helper function to find the containing block (e.g., BlockStatement)
+function findContainingBlock(node) {
+	let current = node;
+	while (current) {
+		if (current.type === "BlockStatement" || current.type === "Program") {
+			return current;
+		}
+		current = current.parent;
+	}
+	return null;
+}
+
+// Helper to resolve variable values in scope
+function getKeyValue(node, context) {
+	if (node.type === "Literal") return node.value;
+	if (node.type === "Identifier") {
+		let scope = context.getScope();
+		while (scope) {
+			const variable = scope.variables.find((v) => v.name === node.name);
+			if (variable) {
+				// Check all definitions for initial values
+				for (const def of variable.defs) {
+					const init = def?.node?.init;
+					if (!init) continue;
+
+					// Handle nested literals (e.g., const key = "test")
+					if (init.type === "Literal") {
+						return init.value;
+					}
+					// Handle simple identifier references (e.g., const key = KEY_CONSTANT)
+					if (init.type === "Identifier") {
+						return getKeyValue(init, context); // Resolve recursively
+					}
+				}
+				break;
+			}
+			scope = scope.upper;
+		}
+		return node.name;
+	}
+	return null;
+}
+
+// Helper to check if else block assigns the key to the base object
```

Comment:
Can we expand on this? A small example of what kind of thing the function validates would be nice.